Clarified prune documentation regarding tags and interpolate

3ygun opened the pull request:
In response to #13 
- [x] Clarified that tags are affected by whitelist and blacklist
- [x] Clarified what interpolate does by displaying the common logstash dynamic value syntax
  - [x] The `%{some_field}` should be in the `code` style < Don't know how to do :/
- [x] Clarified how to specify an exact field or value name

3ygun committed:
Update prune.rb

- [x] Clarified that tags are affected by whitelist and blacklist
- [x] Clarified what interpolate does by displaying the common logstash dynamic value syntax
      - [ ] The %{some_field} should be in a code tag < Don't know how to do :/

magnusbaeck commented on lib/logstash/filters/prune.rb:
```diff
@@ -16,11 +16,11 @@ class LogStash::Filters::Prune < LogStash::Filters::Base
 
   # Trigger whether configuration fields and values should be interpolated for
-  # dynamic values.
+  # dynamic values (when resolving %{some_field}).
```
In Asciidoc you can use backticks (`) to format text as code (which I agree would be appropriate in this case).

magnusbaeck commented on lib/logstash/filters/prune.rb:
```diff
@@ -16,11 +16,11 @@ class LogStash::Filters::Prune < LogStash::Filters::Base
   config :interpolate, :validate => :boolean, :default => false
 
-  # Include only fields only if their names match specified regexps, default to empty list which means include everything.
+  # Include only fields and tags only if their names match specified regexps, default to empty list which means include everything.
```
The way you've reworded this sentence (and the corresponding one on line 32) could give the impression that the pruning acts on tag values themselves rather than the `tag` field, i.e. that you could say `whitelist_names => ["foo"]` to keep the "foo" tag but remove all other tags.

Would it be a good idea to reword this to avoid that ambiguity? Maybe add a sentence (here or elsewhere) similar to "The `tags` field that stores the event's tags is also subject to pruning."

magnusbaeck commented:
If you don't mind terribly, could you amend the commit and update the commit message to not just say "Update prune.rb"? The PR's title ("Clarified prune documentation regarding tags and interpolate") would be much better.

3ygun committed:
Clarified prune documentation regarding exact fields, tags and interpolate

- [x] Add documentation regarding using exact matching a field or value name
- [x] Fixed interpolate formatting with ```%{some_field}``` being in code marks
- [x] Fixed documentation of how the prune filter handles ```tags``` following @magnusbaeck suggested improvements

3ygun commented:
Requested changes made.

magnusbaeck commented:
Thanks! I'll look into making a 2.0.5 release with this PR.

magnusbaeck commented:
> I'll look into making a 2.0.5 release with this PR.

2.0.5 apparently already exists (but hadn't been tagged), so it'll be 2.0.6.
